llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -2028,9 +2028,18 @@ struct DSEState {
     if (!InnerCallee)
       return false;
     LibFunc Func;
+    std::optional<StringRef> ZeroedVariantName = std::nullopt;
```

Comment:
```suggestion
    StringRef ZeroedVariantName;
```
I think it would be a bit simpler to use the empty string as the sentinel here. But I'm ok either way.